### PR TITLE
fix: check both SBUS trainer ports regardless of TRAINER_MODULE_SBUS_USART

### LINKED ISSUE #5139: Trainer SBUS does not work on TARANIS QX7-Access
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

I have a SBUS source connected to the external module bay. After upgrading to 2.10.1 from 2.9.3 there is no trainer input any more. If I switch to a model with "master/SBUS" mode I get neither any haptic feedback nor any trainer input.

### Expected Behavior

Power on external module (haptic feedback). Get trainer input from SBUS.

### Steps To Reproduce

Change to a model with "master/SBUS" mode. Watch for haptic feedback, and TRx values.

### Version

2.10.1

### Transmitter

FrSky Q X7 / Q X7S (ACCESS)

### Operating System (OS)

Linux

### OS Version

Debian Bookworm

### Anything else?

This is a regression error because of the refactoring of the trainer driver code in 2.10. I am working on a fix.

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
Fixes #5139

Summary of changes:
- removed `#if defined(TRAINER_MODULE_SBUS_USART)` and just check presence of potential ports instead.

### CODE REVIEW

**Comment by mha1** [comment]
Thanks

**Comment by bug400** [comment]
Successfully tested with QX7Acces radio (main and 2.10.1)

**Comment by Tiefflieger68** [comment]
@pfeerick

I've struggled about an "issue" with my RM MT12.
For my understanding HB should be used if there is no independend SPort Pin in the Module Bay (shared Pin for internal and external Module). But there is a independent SPort Pin in the MT12 Radio. So SPort should be used for SBUS Trainer.

Find the discussion started by me at Discord:
https://discord.com/channels/839849772864503828/839895574244229152/1257669878924705792

**Comment by raphaelcoeffic** [comment]
> For my understanding HB should be used if there is no independend SPort Pin in the Module Bay (shared Pin for internal and external Module). But there is a independent SPort Pin in the MT12 Radio. So SPort should be used for SBUS Trainer.

It's the opposite: S.PORT pin is only used if no heartbeat pin can be used (no hardware serial). It was mistaken in 2.10.1 and has been fixed in 2.10.2.

**Comment by Tiefflieger68** [comment]
OK - Thanks for clarification

**Comment by mha1** [comment]
@raphaelcoeffic I have yet to check TX16s 2.10.2. It has a hardware serial on HB but as you know not functional for receiving serial data without a hairy mod. Place your bets if the change killed Master/SBUS on the TX16s.

**Comment by 3djc** [comment]
The HB pin on TX16S does not support inversion, so SPORT pin is used there with 2.10.2

**Comment by mha1** [comment]
yes, tested, TX16s works with SBUS on  SPort.

**Comment by Will-Ac** [comment]
Using V2.10.4
I have tried to get Master/SBUS feature working on Radiomaster Zorro without luck. 
I used a Happymodel EP2 RX connected to module bay. 
Before connecting to module bay, EP2 was set to SBUS and confirmed as outputting SBUS using ToolkitRC SBUS input checker, when moving sticks on student (Slave) RM pocket radio.
When I connected EP2 to Instructor radio (Master) module bay, I used a 5V voltage regulator to avoid hurting the EP2, which powered up fine when Master/SBUS was selected as trainer source in model settings and bound to student radio as expected. 
I tried connecting the TX pad on EP2 to both the HB and SPORT pins with no luck.

I also tried using ToolkitRC SBUS output hooked up to  module bay again to both HB and SPORT pins also without luck. 

I also tried the above with Pocket and Zorro reversed. No luck.
Has this feature been tested on either Zorro or Pocket?

I do hope I can get it working as it’s a really nice feature and would be great to have it up &running and fully documented. 

I tried the wired trainer method and all worked fine as expected.

**Comment by Will-Ac** [comment]
I have finally figured it out. :)
TLDR: on Zorro, for SBUS in I had to use Module pin PXX IN rather than SPORT to get it to work.

It appears that different pins may have to be used on different radios.
To make this feature usable the documentation needs to be updated to clarify which pin on which radio, if they are indeed different. and maybe also whether it requires SBUS or Inverted SBUS isignal in ELRS setup.  On my RX I used SBUS.

My working setup is as follows:
EdgeTX V2.10.4
Master setup->
  Radio: Zorro internal ELRS
  Setup in Module bay (right pin1 - left pin8) ->
    Module GND (pin2) VMAIN (pin3) provide 8.4V
    Micro BEC steps 8.4V down to 5V. (I will use smallest available being Matek Micro BEC) mine is a bit bigger.
    Happymodel EP2 receiver wired to BEC (Max 5V input)
    Module PXX IN (pin6) wired EP2 TX pad. (I had wrongly tried SPORT and Heartbeat pins, but these might work on other radios) this needs to be better documented
  External RF Mode: OFF
  Trainer Mode: Master/SBUS
Global functions (or model sepcfic Special functions)->
  On push button -  Trainer - Axis (or Chans)


Pupil Setup ->
  Radio: Pocket internal ELRS
  Internal RF - Mode: CRSF  (ie ELRS)
  Internal module bound to EP2 receiver in Master Radio using different Bind phrase from phase used between Master radio TX and plane RX

**Comment by Will-Ac** [comment]
I found same pinout works on RM Pocket.

Here's is my final hardware setup using smaller RX and DC-DC converter. 

image
